lib/scope.js

```diff
@@ -909,36 +921,73 @@ module.exports = {
-
-      // Feedback for devs - we need the trigger var but we don't have it (field data)
-      // TODO: Add docs and link to docs on how to add the trigger var
-      if ( field.trigger === undefined ) { await scope.addToReport(scope, { type: `warning`, value: `A field on this page uses an index variable or a generic object, but the page does not contain an element with the trigger variable's name. Tell the developer to add an element to the page with the id "trigger"` }); }
+    // Make sure the trigger var matches for a page that uses proxies. Only
+    // story tables need this disambiguation.
+    if ( trigger_might_be_needed && uses_proxies && matches.length > 0 ) {
+
+      if ( !field.trigger ) {
+        // field/page trigger var is needed but is missing.
+        let page_trigger_warning = `A field on this page uses an index `
+          + `variable or a generic object, but the interview page does not `
+          + `give the trigger variable's name. See `
+          + `https://suffolklitlab.org/docassemble-AssemblyLine-documentation`
+          + `/docs/automated_integrated_testing/#a-missing-trigger-variable.`;
+        await scope.addToReport(scope, { type: `warning`, value: page_trigger_warning });
+      }
 
       let original_matches = matches;
+      let fallbacks = [];
       matches = [];
       for ( let match of original_matches ) {
-        // Feedback for devs - we need the trigger var but we don't have it (var data)
-        if ( match.trigger === undefined ) { await scope.addToReport(scope, { type: `warning`, value: `A field on this page uses an index variable or a generic object. The test gave story data for setting the field, but that data is missing the trigger. Add a trigger variable to the story table \`trigger\` column. For now, the test may end up ignoring this field. The row's data is\n${ JSON.stringify( match.original )}.\nThe field's data is\n${ JSON.stringify( field )}` }); }
-          // TODO: Add a warning if they're both `undefined`?
-        else {
-
+        if ( !match.trigger ) {
+          // var/row trigger data is (probably) needed but is missing. Caveat: If there's
+          // only one page in the proxy var loop and there's only one row, trigger var is
+          // not needed, but I'm not sure how we figure out all those things here and log
+          // a useful message.
+          let row_trigger_warning = `This story table row needs the "trigger" `
+            + `value, but it's missing. The test might get unexpected results. See `
+            + `https://suffolklitlab.org/docassemble-AssemblyLine-documentation`
+            + `/docs/automated_integrated_testing/#trigger.`
+            + `\nThe row's data is\n${ JSON.stringify( match.original )}.`
+            + `\nThis field's data is\n${ JSON.stringify( field )}`;
+          await scope.addToReport(scope, { type: `warning`, value: row_trigger_warning });
+          fallbacks.push( match );
```

Comment:
Collecting fallbacks, but not yet using.

We might be able to provide the developer with the variable names and values that we are setting them to.

Reply:
Actually, I forgot the values given to the field will be visible in the info of the report later. Also, I think I misread the code. `\nThe row's data is\n${ JSON.stringify( match.original )}.` shows the data of that particular field. Resolving this for now, but let me know if we meant something different.